lib/fizzy/parser_expr.cpp

```diff
@@ -94,11 +101,71 @@ parser_result<Code> parse_expr(const uint8_t* pos)
         case Instr::i64_extend_i32_u:
             break;
 
+        case Instr::end:
+        {
+            if (!label_positions.empty())
+            {
+                const auto label_pos = label_positions.back();
+                label_positions.pop_back();
+                if (label_pos != loop_label)  // If end of block instruction.
+                {
+                    const auto target_pc = static_cast<uint32_t>(code.instructions.size() + 1);
+                    const auto target_imm = static_cast<uint32_t>(code.immediates.size());
+
+                    // Set the imm values for block instruction.
+                    auto* block_imm = code.immediates.data() + label_pos;
+                    __builtin_memcpy(block_imm, &target_pc, sizeof(target_pc));
```

Comment:
Why can't you use `push` helper here?

Reply:
Ah got it, because it doesn't push, but modifies existing values.

But still would be better to extract this to another helper.